### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by vbmade2000** [noclick-me/noclickd#48] [Q&A] [answered]
I am a novice Rustacean. Please tell me how to contribute.

**Comment by llucax** [comment]
Hi @vbmade2000, thanks for jumping in! It is awesome to that you want to contribute. I converted this to a discussion as it is a more effective channel for this sort of things.

To be honest, I am a novice Rustacean too, and it's been a few months since last time I touched Rust even, so if you want to have some sort of mentor to learn Rust, this might not be the best project :-/

If you are still interested in contributing, you are extremely welcome. My recommendation would be to just pick an issue that interests you and go for it.

If you have any other more specific questions, feel free to ask here (in this thread or if it is more specific you can create a new post).

Thanks again, and welcome!

**Reply by vbmade2000** [reply]
I appreciate the quick reply. 

Actually, it is true that I was looking for some mentor for contribution. But I think I'll still look at the issues if I can find something to contribute to. Thanks, @llucax.

**Reply by vbmade2000** [reply]
Is it just your side-project o you have plans to commercialize it?

**Reply by llucax** [reply]
Both. It is a side project but I have intentions to monetize it eventually, if I ever have the time to resume it. I would love to be able to make a living from developing free software, so I chose to make it open source. If I ever get any money from it, first it will be used to pay for proper server infrastructure to provide a free service (I'm already providing a free instance at https://noclick.me but it is very rudimentary).

**Reply by vbmade2000** [reply]
That's great. Where is the instance hosted currently?

**Reply by llucax** [reply]
GCP free tier

**Reply by vbmade2000** [reply]
Okay. But that will be for some limited time period. Correct?

**Reply by llucax** [reply]
No, it is free *forever*.

**Reply by vbmade2000** [reply]
Hey, is there any doc or something that describes how it works?

**Reply by llucax** [reply]
Sorry, I've been having very little time lately. I will try to write something up and put it in the repo, but it is very simple for now, the basics is just a hash table that maps URL IDs/slugs to the real URL. That's pretty much it. All the rest is Rust and framework stuff.

**Reply by llucax** [reply]
Again sorry @vbmade2000. I guess by now you (understandably) moved to some other project, but just in case I wanted to say sorry again for never writing anything up. It was (and still is) a very busy time and even when I would love to spend some time in this project, it won't happen in the short term.

**Comment by vbmade2000** [comment, marked as answer]
Hey @llucax, I am extremely sorry. I haven't moved to other project but got into bit of a work in current job. Hopefully, I'll take this issue after some times. Again, apology.
